coordinator FAQ (#68)
hermeznetwork/docs: docs/faq/coordinators.md

#  Coordinators

## Overview

### What is a Hermez Coordinator?

> A Coordinator is our term for rollup block producer. At any one time there is one Coordinator responsible for creating blocks on the rollup chain selected among a number of registered nodes via an auction process.

Reviewer: ```suggestion
A Coordinator is our term for rollup block producer. At any one time, there is one Coordinator responsible for creating blocks on the rollup chain selected among a number of registered nodes via an auction process.
```

### How many Coordinators are there?

There is no limit to the number of registered Coordinators. Becoming a Coordinator is entirely permissionless via an auction process and it will be enabled shortly after launch.

> Although the first coordinator will be the Hermez Boot Coordinator which will forge blocks when there's no alternative bids in the auction, it’s important for us that the market for coordinators becomes open over time.

Reviewer: ```suggestion
Although the first coordinator will be the Hermez Boot Coordinator which will forge blocks when there are no alternative bids in the auction, it’s important for us that the market for coordinators becomes open over time.
```

### How is the Coordinator Node selected?

The Coordinator Node is selected using an auction process, where registered nodes bid for the right to forge batches during a time slot. The highest bidder in a given slot will become the Coordinator

### Will there be a competitive market for Coordinators?

> We are committed to creating a competitive market for coordinators and we will open source software to allow anyone to run a coordinator.

Reviewer: ```suggestion
We are committed to creating a competitive market for Coordinators and we will open-source software to allow anyone to run a coordinator node.
```

## Auction Process

### How do I become a Coordinator?

> To become a coordinator you need to prepare the system infrastructure, take part in the auction and win a bid for a time slot.

Reviewer: ```suggestion
To become a coordinator you need to prepare the system infrastructure, take part in the auction, and win a bid for a time slot.
```

### How bids can be placed in the auction?

> Coordinators will participate in the auction by sending an on-chain transaction to the auction smart contract. Bids are always payed in HEZ

Reviewer: ```suggestion
Coordinators will participate in the auction by sending an on-chain transaction to the auction smart contract. Bids are always paid in HEZ.
```

### How do Coordinators make money?

Coordinators set and collect the transaction fees. They can expect a revenue per transaction and each coordinator will select to forge the more profitable transactions from the transaction pool.

They profit from these fees minus the operational costs and the bid price for the slot.

## Running a Coordinator

### What are the infrastructure requirements to run a Coordinator?

To run a Coordinator you need the following:
- PostgreSQL database
- Ethereum Node
- Coordinator Server running the [hermez-node repository](https://github.com/hermeznetwork/hermez-node). A reasonable server spec is AWS instance c5ad.2xlarge with 8 vCPU and 16GB RAM
- Proof Server running the [rapidsnark repository](https://github.com/iden3/rapidsnark). A server spec able to run the ~2000 transaction circuit is AWS instance c5ad.24xlarge with 96 vCPU and 192GB RAM. 

### Is there a step by step guide to configure a Coordinator node?

Yes. You can find this guide [here](../developers/coordinator.md).

### How many transactions can be processed per batch?
 

> Number of transactions per batch depend on the circuit used. Hermez accepts two different circuit sizes: 344 transactions and 1912 transactions. 

Reviewer: Additional questions to consider adding:

1.) I'm having technical issues ("as a Coordinator" or "running a coordinator node")  who can I contact for support?
2.)Is there requirement for the number of HEZ tokens you have to hold to be a Coordinator? 
3.) Will there be a public registration of the Hermez Network Coordinators?

#  Coordinators

## Overview

### What is a Hermez Coordinator?
A Coordinator is our term for rollup block producer. At any one time, there is one Coordinator responsible for creating blocks on the rollup chain selected among a number of registered nodes via an auction process.

### How many Coordinators are there?

There is no limit to the number of registered Coordinators. Becoming a Coordinator is entirely permissionless via an auction process and it will be enabled shortly after launch.

> Although the first coordinator will be the Hermez Boot Coordinator which will forge blocks when there are no alternative bids in the auction, it’s important for us that the market for coordinators becomes open over time.

Reviewer: ```suggestion
Although the first Coordinator will be the Hermez Boot Coordinator, which will forge blocks when there are no alternative bids in the auction, it’s important for us that the market for coordinators becomes open over time.
```

### How is the Coordinator Node selected?

The Coordinator Node is selected using an auction process, where registered nodes bid for the right to forge batches during a time slot. The highest bidder in a given slot will become the Coordinator

### Will there be a competitive market for Coordinators?

We are committed to creating a competitive market for Coordinators and we will open-source software to allow anyone to run a Coordinator node.

## Auction Process

### How do I become a Coordinator?

> To become a coordinator you need to prepare the system infrastructure, take part in the auction, and win a bid for a time slot.

Reviewer: ```suggestion
To become a Coordinator you need to prepare the system infrastructure, take part in the auction, and win a bid for a time slot.
```

> ### How bids can be placed in the auction?

Reviewer: ```suggestion
### How are bids placed in the auction?
```

If the Coordinator has not done anything in the first part a slot, then anyone can jump in and replace it by forging blocks (first come first served).

### How do Coordinators make money?

Coordinators set and collect the transaction fees. They can expect a revenue per transaction and each coordinator will select to forge the more profitable transactions from the transaction pool.

They profit from these fees minus the operational costs and the bid price for the slot.

## Running a Coordinator

### What are the infrastructure requirements to run a Coordinator?

To run a Coordinator you need the following:
- PostgreSQL database
- Ethereum Node
- Coordinator Server running the [hermez-node repository](https://github.com/hermeznetwork/hermez-node). A reasonable server spec is AWS instance c5ad.2xlarge with 8 vCPU and 16GB RAM
- Proof Server running the [rapidsnark repository](https://github.com/iden3/rapidsnark). A server spec able to run the ~2000 transaction circuit is AWS instance c5ad.24xlarge with 96 vCPU and 192GB RAM. 

### Is there a step by step guide to configure a Coordinator node?

Yes. You can find this guide [here](../developers/coordinator.md).

> ### How do you register a Coordinator?

Reviewer: ```suggestion
### How do you register as a Coordinator?
```

You can use a JavaScript tool, [cli-bidding](https://github.com/hermeznetwork/cli-bidding) to register your node as a Coordinator.

### How many transactions can be processed per batch?
 

> Number of transactions per batch depend on the circuit used. Hermez accepts two different circuit sizes: 344 transactions and 1912 transactions. 

Reviewer: ```suggestion
The number of transactions per batch depends on the circuit used. Hermez accepts two different circuit sizes: 344 transactions and 1912 transactions. 
```
